nexusdialog/src/main/java/com/github/dkharrat/nexusdialog/controllers/CheckBoxController.java

```diff
@@ -132,7 +140,7 @@ public void refresh(CheckBox checkbox, int index) {
         Set<Object> modelValues = retrieveModelValues();
         checkbox.setChecked(
                 modelValues.contains(
-                        areValuesDefined() ? checkbox.getText() : index
+                        !areValuesDefined() ? checkbox.getText() : values[index]
```

Comment:
Okay, basically the problem is that my values are not my text of the checkbox, since sometimes I might have a checkbox with value a and translated title of a. I see your point that the 0-based index muss be supported, so I updated the PR